app/controllers/projects_controller.rb

```diff
@@ -63,20 +63,28 @@ def index
                      current_power.samples
                    end
 
-        @projects = if only_library || exclude_library
-                      @samples.group(:project).count
+        # Retrieve a json of projects associated with samples;
+        # augment with number_of_samples, hosts, tissues.
+        # TODO: Why are we passing host / tissue for each individual sample, with a lot of duplication?
```

Comment:
We are not passing per sample... it is grouped below.
This was made more complex due to the inability to access samples from projects... something we should revisit as we review our access control logic and/or methodology.

Reply:
But it passes something like `{tissues: [plasma, plasma, plasma, csf, serum, plasma, plasma,...]}`, right? (Correct me if I'm wrong.) Just curious if we could pass something like `{tissues: {plasma: 5, csf: 1, serum: 1}}` instead. Anyway, that would be for future PR...

Reply:
Oh never mind -- I was missing the `distinct`. Thanks, something I need to fix in my change

Reply:
I think the way I will test my change is that I'll explicitly write a function to compare the old and the new output for each action, and then I'll post the results here -- so I'm not missing any more things like this